### DISCUSSION TITLE
Undefined Ghost haunts me!!

### DISCUSSION THREAD

**Discussion by ostermann-felix** [neuefische/questions#230] [Web]
Moin Freunde, 

ich hab gerade ein undefined Problem. Als ich gerade meine Index.js meiner App auf eine Komponentendatei ausgelagert habe ist folgender Fehler aufgetreten: 

<img width="362" alt="Bildschirmfoto 2022-10-08 um 19 53 23" src="https://user-images.githubusercontent.com/94837451/194720984-1d9f0cfd-7831-4e13-8865-1a2e8b79dce4.png">

In der Komponente wird ein Fetch meiner Datenbank umgemapt und als Links ausgegeben. Wenn man drauf klickt soll man dann auf die entsprechende Kategorieseite kommen. Die Kategorie ist in jedem einzelnen Datenbank Eintrag vermerkt mit Bild und Name. 

Der PR dazu: https://github.com/ostermann-felix/capstone-project/pull/10
ThankXXXX XoXo

**Comment by manusmd** [comment]
Du hast dir durch deine getServerSideProps Funktion deine Daten ein bisschen arg verschlachtelt.
Probiers mal über dem return in der Komponente damit:
```

export default function CategoryGrid() {
  const [lessons, setLessons] = useState([]);

  useEffect(() => {
    const getLessons = async()=> {
      const response = await getAllLessons();
      setLessons(response); 
    }
    getLessons();
  }, [])
```
So hast du bei jedem rendern der CategoryGrid Komponente die aktuellen lessons. Falls du die lessons weitehin von außen in die Komponente geben willst würdest du die paar Zeilen in die Eltern Komponente packen und den state (lessons) als prop in die Komponente übergeben.
